add_keys.py

```diff
@@ -0,0 +1,204 @@
+#!/usr/bin/env python3
+
+import os
+import json
+import logging
+import sys
+import time
+import click
+from web3 import Web3, WebsocketProvider, HTTPProvider
+from web3.exceptions import SolidityError, CannotHandleRequest, TimeExhausted
+
+logger = logging.getLogger().setLevel("INFO")
+
+ARTIFACTS_DIR = './assets'
+POOL_ADDRESS = "0xae7ab96520de3a18e5e111b5eaab095312d7fe84"
+POOL_ARTIFACT_FILE = 'Lido.json'
+REGISTRY_ARTIFACT_FILE = 'NodeOperatorsRegistry.json'
+GAS_LIMIT=13500000
+
+
+def init_provider(eth1_provider):
+    '''
+    Initialise the Web3 provider.
+    '''
+    if eth1_provider.startswith('http'):
+        provider = HTTPProvider(eth1_provider)
+    elif eth1_provider.startswith('ws'):
+        provider = WebsocketProvider(eth1_provider)
+    else:
+        logging.error('Unsupported ETH provider!')
+        sys.exit(1)
+
+    return Web3(provider)
+
+
+def init_contracts(w3: Web3, pool_address):
```

Comment:
Great, type annotations! :blush: would be nice to add them to more arguments and return types. Then Mypy can also perform some basic sanity checks.